lua/pixelui/drawing/cl_imgur.lua

```diff
@@ -17,27 +17,32 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
 ]]
 
 local progressMat
-local getImgur = PIXEL.GetImgur
-getImgur("635PPvg", function(mat) progressMat = mat end)
 
-local curTime = CurTime
+local drawProgressWheel
 local setMaterial = surface.SetMaterial
 local setDrawColor = surface.SetDrawColor
-local drawTexturedRectRotated = surface.DrawTexturedRectRotated
-local drawTexturedRect = surface.DrawTexturedRect
 
-local min = math.min
-local function drawProgressWheel(x, y, w, h, col)
-    local progSize = min(w, h)
-    setMaterial(progressMat)
-    setDrawColor(col.r, col.g, col.b, col.a)
-    drawTexturedRectRotated(x + w / 2, y + h / 2, progSize, progSize, -curTime() * 100)
+do
+    local min = math.min
+    local curTime = CurTime
+    local drawTexturedRectRotated = surface.DrawTexturedRectRotated
+
+    function PIXEL.DrawProgressWheel(x, y, w, h, col)
+        local progSize = min(w, h)
+        setMaterial(progressMat)
+        setDrawColor(col.r, col.g, col.b, col.a)
+        drawTexturedRectRotated(x + w * .5, y + h * .5, progSize, progSize, -curTime() * 100)
+    end
+    drawProgressWheel = PIXEL.DrawProgressWheel
 end
-PIXEL.DrawProgressWheel = drawProgressWheel
 
 local materials = {}
 local grabbingMaterials = {}
 
+local getImgur = PIXEL.GetImgur
+getImgur("635PPvg", function(mat) progressMat = mat end)
```

Comment:
This should be configurable to some degree. Also the callback should be on another line like so:
```lua
getImgur("635PPvg", function(mat) 
    progressMat = mat 
end)
```

Reply:
Agreed, I'll make a more uniform configuration system in a different PR.